data_diff/utils.py

```diff
@@ -51,3 +52,9 @@ def number_to_human(n):
     )
 
     return "{:.0f}{}".format(n / 10 ** (3 * millidx), millnames[millidx])
+
+
+def remove_password_from_url(url, replace_with="***"):
+    parsed = urlparse(url)
+    replaced = parsed._replace(netloc="{}:{}@{}".format(parsed.username, replace_with, parsed.hostname or ""))
```

Comment:
**Important 1:** It loses the port value here if it is set, which is often essential for users.

**Important 2:** It might inject the `***` even if no password was there at all.

Similarly, it will inject the literal string `None` in place of a username if no username was specified there originally, or maybe it will insert an unnecessary `:@` prefix — e.g. when using different auth methods than the "username:password" pair; maybe even with tokens.

Reply:
You're right, I can do it better. thanks!